tools/collect_hana_sr_diag.sh

```diff
@@ -0,0 +1,648 @@
+#!/bin/bash
+set -e
+#set -x
+
+# --- GLOBAL ---
+SSH_USER=cloudadmin # eventually overwritten by in  parse_inventory_file
+
+# --- Script Setup ---
+# Function to parse config.yaml
+parse_config_file() {
+    local config_file="$1"
+    local sid=""
+    local instance_num=""
+    local provider_val=""
+
+    if [[ -f "$config_file" ]]; then
+        sid=$(grep "sap_hana_install_sid:" "$config_file" | grep -v '^[[:space:]]*#' | head -n 1 | sed -e 's/.*sap_hana_install_sid:[[:space:]]*//' -e "s/['\"]//g" -e 's/#.*//' | xargs)
+        instance_num=$(grep "sap_hana_install_instance_number:" "$config_file" | grep -v '^[[:space:]]*#' | head -n 1 | sed -e 's/.*sap_hana_install_instance_number:[[:space:]]*//' -e "s/['\"]//g" -e 's/#.*//' | xargs)
+        # The following command extracts the provider value from the YAML config_file.
+        # It handles values with or without quotes (e.g., 'azure', "azure", azure),
+        # trims whitespace, and ignores commented lines.
+        provider_val=$(grep "provider:" "$config_file" | grep -v '^[[:space:]]*#' | head -n 1 | sed -e 's/.*provider:[[:space:]]*//' -e "s/['\"]//g" -e 's/#.*//' | xargs)
```

Comment:
I still think there's a way to use fewer commands in these pipelines. For example, for the first one perhaps something like:
```
sid=$(sed -n -e 's/#.*//' -e "s/['\"]//g" -e 's/.*sap_hana_install_sid:[[:space:]]*//p' "$config_file")
```
Would be enough.